workflow/kfdrc_production_mutect2_wf.cwl

```diff
@@ -0,0 +1,155 @@
+inputs:
+  # Required
+  reference_fasta: { type: File }
+  reference_fai: { type: 'File?' }
+  reference_dict: { type: 'File?' }
+  input_tumor_aligned:
+    type: File
+    secondaryFiles: |
+      ${
+        var dpath = self.location.replace(self.basename, "")
+        if(self.nameext == '.bam'){
+          return {"location": dpath+self.nameroot+".bai", "class": "File"}
+        }
+        else{
+          return {"location": dpath+self.basename+".crai", "class": "File"}
+        }
+      }
+    doc: "tumor BAM or CRAM"
+  input_tumor_name: string
+  input_normal_aligned:
+    type: File
+    secondaryFiles: |
+      ${
+        var dpath = self.location.replace(self.basename, "")
+        if(self.nameext == '.bam'){
+          return {"location": dpath+self.nameroot+".bai", "class": "File"}
+        }
+        else{
+          return {"location": dpath+self.basename+".crai", "class": "File"}
+        }
+      }
+    doc: "normal BAM or CRAM"
+  input_normal_name: string
+  vep_cache: { type: File, doc: "tar gzipped cache from ensembl/local converted cache" }
+  mutect2_af_only_gnomad_vcf: { type: File }
+  mutect2_af_only_gnomad_tbi: { type: 'File?', doc: "Tabix index for mutect2_af_only_gnomad_vcf" }
+  mutect2_exac_common_vcf: { type: File }
+  mutect2_exac_common_tbi: { type: 'File?', doc: "Tabix index for mutect2_exac_common_vcf" }
+  output_basename: { type: string, doc: "String value to use as basename for outputs" }
+  wgs_or_wxs: { type: { type: enum, name: sex, symbols: ["WGS", "WXS"] }, doc: "Select if this run is WGS or WXS" }
+
+  # Optional with One Default
+  select_vars_mode: { type: ['null', { type: enum, name: select_vars_mode, symbols: ["gatk", "grep"] }], default: "gatk", doc: "Choose 'gatk' for SelectVariants tool, or 'grep' for grep expression" }
+  vep_ref_build: { type: 'string?', default: "GRCh38", doc: "Genome ref build used, should line up with cache" }
+
+  # Optional with Multiple Defaults (handled in choose_defaults)
+  exome_flag: { type: 'string?', doc: "Whether to run in exome mode for callers. Y for WXS, N for WGS" }
+
+  # WGS only Fields
+  wgs_calling_interval_list: { type: 'File?', doc: "GATK intervals list-style, or bed file.  Recommend canocical chromosomes with N regions removed" }
+
+  # WXS only Fields
+  padded_capture_regions: { type: 'File?', doc: "Recommend 100bp pad, for somatic variant" }
+
+outputs:
+  mutect2_vep_vcf: { type: File, outputSource: run_mutect2/mutect2_vep_vcf }
+  mutect2_vep_tbi: { type: File, outputSource: run_mutect2/mutect2_vep_tbi }
+  mutect2_prepass_vcf: { type: File, outputSource: run_mutect2/mutect2_filtered_vcf }
+  mutect2_vep_maf: { type: File, outputSource: run_mutect2/mutect2_vep_maf }
+
+steps:
+  choose_defaults:
+    run: ../tools/mode_defaults.cwl
+    in:
+      input_mode: wgs_or_wxs
+      exome_flag: exome_flag
+    out: [out_exome_flag,out_cnvkit_wgs_mode,out_i_flag,out_lancet_padding,out_lancet_window,out_vardict_padding]
```

Comment:
no change required, but CWL does not mandate that all outputs from a task actually need to be declared in a parent workflow, in case in the future you wanted to shorten this up and leave out cnvkit, lancet, vardict etc outputs